### PR TITLE
Add support for Base64DecodeExt

### PR DESCRIPTION
**Pull request by soujanyanmbri**
> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Make sure that you've checked the boxes below before you submit PR:**

- [ ✅ ] My code includes positive and negative tests.
- [✅ ] I have an appropriate description with correct grammar.
- [ ✅] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v3sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).
- [✅ ] My code is properly linted and passes pre-commit tests.

Thanks for your contribution :heart:

This PR does not need any more tests since they are already covered in `internal/transformations/testdata/base64DecodeExt.json`

### CODE REVIEW

**Comment by jptosso** [comment]
lgtm, looking for another approval

**Review by fzipi — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Would like to see this in the main documentation repo also, so people can start using it.

Can you please add tests? Both positive and negative, as the PR template says ;)

**Review by fzipi — COMMENTED** [review]
Can you also provide a reasonable explanation on *why* this change? Which kind of traffic would use this instead of the regular one?

**Comment by soujanyanmbri** [comment]
> Can you also provide a reasonable explanation on _why_ this change? Which kind of traffic would use this instead of the regular one?
This is to maintain consistency with modsec with base64DecodeExt operator. base64DecodeExt is mostly to take care of base64decode evasions while the payloads are used in PHP machines (Since, PHP accepts . or " " in base64encoded payloads while decoding) 
https://github.com/owasp-modsecurity/ModSecurity/wiki/Reference-Manual-%28v3.x%29#user-content-base64DecodeExt

**Comment by fzipi** [comment]
Awesome, thanks for documenting this.

**Comment by jcchavezs** [comment]
Thanks for this <3. I would propose a couple of things here:
1. Not pass more booleans but instead add a build tag for this and if the benchmarks are in favour of this new implementation use that by default.
2. Show the benchmark results in the description of the PR assesing why is this better than the existing one.

**Comment by jptosso** [comment]
@jcchavezs @fzipi 

image

https://github.com/owasp-modsecurity/ModSecurity/wiki/Reference-Manual-%28v2.x%29#base64decodeext

We don't need another reason, it's for compatibility.

File: internal/transformations/base64decode.go [hunk 1]
```diff
@@ -29,11 +32,11 @@ var base64DecMap = []byte{
 // would be possible to use the standard library only relying on undocumented behaviors of the decoder.
 // For more context, see https://github.com/corazawaf/coraza/pull/940
 func base64decode(data string) (string, bool, error) {
-	res := doBase64decode(data)
+	res := doBase64decode(data, false)
 	return res, true, nil
 }
 
-func doBase64decode(src string) string {
+func doBase64decode(src string, ext bool) string {
```
Comment (jcchavezs) [inline internal/transformations/base64decode.go:39]: name is meaningful. I would use typed constants for this or at least add a comment on what ext is.

**Comment by jcchavezs** [comment]
Thanks for the context @jptosso

File: internal/transformations/base64decode.go (hunk 1 above)
Comment (soujanyanmbri) [reply internal/transformations/base64decode.go:40]: I have added a comment regarding the flag.

**Review by fzipi — APPROVED** [review]
